File: frontend/src/Operaciones.js
import React, { Component } from "react";
import axios from "axios"

class Operaciones extends Component {
    constructor(){
        super();
        this.state = {
            primerNumero: "",
            segundoNumero: "",
            resultado: ""
        };
    }


    setPrimerNumero = e =>{
        const { value } = e.target
        this.setState({
            primerNumero : value
        })
    }

    setSegundoNumero = e =>{
        const { value } = e.target
        this.setState({
            segundoNumero : value
        })
    }

    handleSumbit = e =>{
        e.preventDefault();
        axios
            .post("http://localhost:8080/operaciones/suma", {primerNumero: this.state.primerNumero,segundoNumero: this.state.segundoNumero})
            .then((response) => {
                this.setState({resultado: response.data.resultado})
            })
            .catch(err => console.log(err));
    }

    render() {
        return(

            <div className="factorial">
                <div className="card" style={{ maxWidth: '22rem'}}>
                    <div className="card-body" onSubmit={this.handleSumbit} >
                        <h5 className="card-tittle text-center mb-4"> Suma </h5>
                        <form>
                            <div className="form-group">
                                <label> Primer Numero </label>
                                <input className="form-control" placeholder="Numero" defaultValue={this.state.primerNumero} onChange={this.setPrimerNumero} />
                            </div>
                            <div className="form-group">
                                <label> Segundo Numero </label>
                                <input className="form-control" placeholder="Numero" defaultValue={this.state.segundoNumero} onChange={this.setSegundoNumero} />
                            </div>
                            <div className ="form-group">
                                <label> Resultado </label>  
                                <input className="form-control" placeholder="Resultado" defaultValue={this.state.resultado} disabled />
                            </div>

                            <button type="submit" className="btn btn-primary"> Calcular </button>
                        </form>          
                    </div>     
                </div>
            </div>
        );
    }
}

export default Operaciones;

File: backend/src/main/java/com/mingeso/lab1/models/Operaciones.java
package com.mingeso.lab1.models;

public class Operaciones {
    public int primerNumero;
    public int segundoNumero;
    public int resultado;

    public Operaciones(int primerNumero, int segundoNumero, int resultado) {
        this.primerNumero = primerNumero;
        this.segundoNumero = segundoNumero;
        this.resultado = resultado;
    }

    public int getPrimerNumero() {
        return primerNumero;
    }

    public void setPrimerNumero(int primerNumero) {
        this.primerNumero = primerNumero;
    }

    public int getSegundoNumero() {
        return segundoNumero;
    }

    public void setSegundoNumero(int segundoNumero) {
        this.segundoNumero = segundoNumero;
    }

    public int getResultado() {
        return resultado;
    }

    public void setResultado(int resultado) {
        this.resultado = resultado;
    }

    public void suma(){
        resultado = primerNumero + segundoNumero;
    }

    public void resta(){
        resultado = primerNumero - segundoNumero;
    }
}